### DISCUSSION TITLE
Timecode in file name of cut and merged video

### DISCUSSION THREAD

**Discussion by reaseno** [mifi/lossless-cut#2078] [Q&A]
When I cut a video into several parts and merge them, I get a file with a timecode in the file name that I can't make sense of. I would like to have the information of the cut points. Is it possible to generate this again?

**Comment by mifi** [comment]
hi. you can change the merged output name in the export confirmation dialog (lower right)

**Comment by reaseno** [comment]
My question is: What does the number mean? Or is it just a random number. I would like to document the time codes of the cuts. Maybe in the metadata if the file name gets too long. Manual naming would be too cumbersome for me.

**Reply by mifi** [reply]
The number is Unix time. You can store the times of the cuts as chapters in the output file if you select that option

**Comment by reaseno** [comment]
If I cut out individual parts from a video, the file name of the new file contains the information about the cut point. So you know which part of the original video it is. This information is lost if I cut out several parts from a video and then merge them. I can't see how chapters are supposed to preserve this information. Because the new video, which only contains the parts I need, has a completely different running time to the old one.

**Reply by mifi** [reply]
not sure how you want to store this information in the new file but if you enable "Create chapters from merged segments (slow)" it will be stored as chapters in the merged file.

**Reply by reaseno** [reply]
Chapters still don't make sense to me. Only if you want to see where the cut sections start in the new video. But that's not my point. I want to document the start and end cut times of all the segments from the original. So that it could be cut again from the original if necessary. It is also important to know which segments I have taken over from the original. This could be saved as a comment in the container's metadata.

**Reply by mifi** [reply]
what do you need the data for? if you need it programmatically, you could simply just save/export the `.llc` file along with your output, as i believe it contains this data

**Reply by reaseno** [reply]
I need the data so that I can cut from the original again if necessary. But the most important thing is that I want to document which sections I have taken from the original. I don't want an additional file as it is possible to save this in the files metadata. For individual files, it is saved in the file name, which is really very good but no longer practical for merged cuts.

**Reply by mifi** [reply]
If you have an idea how to store this data inside a mkv, mp4, mov, wav, mp3 etc, using ffmpeg, then I'm open to looking into it.

**Reply by reaseno** [reply]
It is explained here:
https://wiki.multimedia.cx/index.php?title=FFmpeg_Metadata

Could be saved under the field comment as it seems tools like MediaInfo ignore custom field names.

**Reply by mifi** [reply]
Have to do some experimentation then to see how much data can actually be stored in a meta field. Seems it's 256 bytes for some formats: https://trac.ffmpeg.org/ticket/4833

And have to decide:
- what to store in this field?
  - start/end time?
  - label?
  - tags?
- how to format the data? like this `00:01:02.03-00:02:00.00,00:05:00.00-00:06:00.00` (2 segments example with only times)?
- what should happen if the data is longer than 256 byes (e.g. too many segments or long label / many tags)? just truncate?

**Reply by reaseno** [reply]
> like this 00:01:02.03-00:02:00.00,00:05:00.00-00:06:00.00

That sounds good. And if it is truncated because of the limited field size, then that can't be changed. I don't normally cut that many parts either.

**Reply by mifi** [reply]
i've created a feature request #2096
